### PR TITLE
fix table cleanup

### LINKED ISSUE #23: drop_table_without_temporal doesn't exist.
`remove_temporal_table` calls `drop_table_without_temporal`, but that method is not defined.

### PR DESCRIPTION
**Pull request by doits**
- correctly remove temporal table
- remove functions, too

fixes #23

### CODE REVIEW

**Comment by doits** [comment]
I just added a spec for it and refactored the database support a little bit. I'm not sure why it was namespaced as before, I can change it back if you think it should be.

I got my hands dirty in writing helpers to check if a function and trigger exists in postgresql. I am not sure if this is the cleanest way but this is what I came up with :-)

For mysql though I didn't, because I have no database to test. Could some of the maintainers of the gem maybe try to implement it? Otherwise I will take a look when I find some more time, though I have to install mysql first etc.

File: spec/support/database.rb [hunk 1]
```diff
@@ -1,13 +1,45 @@
...
+  case adapter_name
+  when 'postgresql'
+    begin
+      ActiveRecord::Base.connection.execute("select(pg_get_functiondef('#{name}'::regprocedure))")
+      true
+    rescue ActiveRecord::StatementInvalid
+      false
     end
+  when 'mysql'
+    raise NotImplementedError, 'TODO: Implement for mysql'
```
Comment (doits) [inline spec/support/database.rb:27]: Can somebody with more mysql experience maybe provide a way how to check if a function is defined?

File: spec/support/database.rb [hunk 2]
```diff
@@ -1,13 +1,45 @@
...
 end
+
+def trigger_exists?(name)
+  case adapter_name
+  when 'postgresql'
+    ActiveRecord::Base.connection.execute(
+      "select (pg_get_triggerdef(oid)) FROM pg_trigger WHERE tgname = '#{name}'"
+    ).first.present?
+  when 'mysql'
+    raise NotImplementedError, 'TODO: Implement for mysql'
```
Comment (doits) [inline spec/support/database.rb:40]: ... same for trigger: Can somebody with more mysql experience maybe provide a way how to check if a trigger is defined?

File: spec/support/database.rb [hunk 3]
```diff
@@ -1,13 +1,45 @@
 # frozen_string_literal: true
 
-module TemporalTables
-  module DatabaseAdapter
-    def self.adapter_name
```
Comment (doits) [inline spec/support/database.rb:5]: Not sure if this namespace is really needed or not, I removed it while refactoring (it wasn't used anywhere afaik).

File: spec/support/database.rb (hunk 2 above)
Comment (doits) [reply spec/support/database.rb]: I got this one working for mysql

File: spec/support/database.rb (hunk 1 above)
Comment (doits) [reply spec/support/database.rb]: This is not needed, since no function is created with mysql adapter. So it can stay not implemented until it is needed.

**Comment by doits** [comment]
There were still some errors for the mysql adapter, because it doesn't create any function but postgresql does. I fixed the code and implemented the trigger check for mysql, too, so now the specs are green and everything should be ready for review here!

Also also marked my other PR as drafts which aren't ready yet (specs missing), so there's no need to review it yet. Some of them are ready for review though: #37, #39, #40 and #43

**Comment by bkroeker** [comment]
Thank you!

**Pull request merged**